templates/common/upgrade/upgrade.go

```diff
@@ -205,26 +277,38 @@ func Upgrade(ctx context.Context, p *Params) (_ bool, rErr error) {
 		oldManifest:     oldManifest,
 		newManifest:     newManifest,
 	}
-	if err := mergeTentatively(ctx, commitParams); err != nil {
-		return false, err
+	actionsTaken, err := mergeTentatively(ctx, commitParams)
+	if err != nil {
+		return Result{}, err
 	}
 
-	return true, nil
+	conflicts, nonConflicts := partitionConflicts(actionsTaken)
+
+	return Result{
+		Conflicts:    conflicts,
+		NonConflicts: nonConflicts,
+	}, nil
 }
 
 // mergeTentatively does a dry-run commit followed by a real commit.
 //
 // We do a dry run first to try to detect any problems before we start mutating
 // the output directory. We'd like to avoid leaving a mess in the output
 // directory if the operation fails.
-func mergeTentatively(ctx context.Context, p *commitParams) error {
+func mergeTentatively(ctx context.Context, p *commitParams) ([]ActionTaken, error) {
+	var actionsTaken []ActionTaken
 	for _, dryRun := range []bool{true, false} {
-		if err := commit(ctx, p, dryRun); err != nil {
-			return err
+		var err error
+		actionsTaken, err = commit(ctx, p, dryRun)
+		if err != nil {
+			return nil, err
 		}
+		sort.Slice(actionsTaken, func(i, j int) bool {
+			return actionsTaken[i].Path < actionsTaken[j].Path
+		})
```

Comment:
Shouldn't this be outside of the for look?

Reply:
Yep, done.